### PR TITLE
Adds no-alert rule

### PR DESCRIPTION
**Pull request by wik-z**
## Suggested rule/changes(s):
* `<no-alert>` 

## Reason for addition/amendment
Disallows alert() function

@netsells/frontend - Please review

### CODE REVIEW

**Comment by rowbi-wan** [comment]
@wik-z should be a warning in my opinion as some people do sometimes use Alert Driven Development.

**Comment by BeccaAnderton** [comment]
@rowancollins89 people should probably not be doing Alert Driven Development 🤔

**Comment by samturrell** [comment]
@BeccaAnderton18 probably not, but it's a solid technique when at wits end 😂

**Comment by samturrell** [comment]
@jdtjenkins Possibly! Could be possible to use something like https://www.npmjs.com/package/dotenv to read the env and then return accordingly

**Comment by wik-z** [comment]
@samturrell I think @jdtjenkins wanted to say is that it should throw a warning when the NODE_ENV is set to development and a warning when it's in production. I agree with that!

**Comment by samturrell** [comment]
Ah, sorry node env, not normal env. In which case yes, you definitely can `process.env.NODE_ENV` i believe

**Comment by samturrell** [comment]
@jdtjenkins Yep, as per above. However let's try not to use laravel mix as examples and dependencies as this package should be build tool agnostic. However NODE_ENV _should_ be available in all instances in some regard, otherwise fallback to dev

**Comment by wik-z** [comment]
Just added an 'Utils' class, with a 'isProduction' method which basically checks if we're in the production env. Tested this on node 6.5 and it worked like a charm, so if it doesn't work for you, it might be worth checking which node version you're using. 

The first idea was to make it a simple function:
```
export function isProduction() {
    return process.env.NODE_ENV === 'production';
}
```

but because we can't use `import`/`export` syntax with CommonJS, I would have to export an object of functions:

```
module.exports = {
    isProduction: function() {
        return process.env.NODE_ENV === 'production';
    }
}
```

This would get really messy in the future though, so the better option is to define the functions separately in this module and then just put them in an object we want to export:

```
function isProduction() {
    return process.env.NODE_ENV === 'production';
}

module.exports = {
    isProduction,
};
```

and that would work, especially that #5 wasn't approved. However, it looks like Node supports classes, so we tried this option, which I personally think is a better and a cleaner practice. It would be great to hear what people think though.

**Comment by rowbi-wan** [comment]
yeah think this should be merged first @samturrell then we can adjust other PRs to allow warnings in dev but errors in prod

**Comment by samturrell** [comment]
Personally I'm a bigger fan of the functional approach for helper/utils, but i'll leave that up to everyone else to decide. (classes also provide a lot more overhead than a functional return and can be harder to share logic between multiple classes) I don't want this to be used too much though for other rules as it will be an absolute minefield on older codebases, and people will just never be able to push code again until hours have been spent fixing linter issues. I think errors should be reserved for things that DEFINITELY shouldn't make it into production. Stuff that users will see. Only real contenders for it would be alerts and debuggers, potentially console statements to a lesser extent.

**Comment by samturrell** [comment]
Again though, i think it's only really useful for those few rules that stop user-facing issues e.g. alerts, debuggers, console.

**Comment by rowbi-wan** [comment]
@jdtjenkins @wik-z whats happening with this PR?

**Comment by wik-z** [comment]
Yea i'll leave the first idea.
If people could just vote which way to do it: class or functions?

Currently the `isProduction` method (and all future methods) is in the Utils class (see code for reference), but we can just export/import the utils functions separately from the file if we want, using module.export.

functions example:
```
module.exports.isProduction = () => {
   return someLogic;
}
```


Upvote for class, downvote for functions

File: readme.md [hunk 1]
```diff
@@ -347,7 +347,17 @@ if (additionalPosts.length) {
 }
 ```
 
----
+#### 📍 no-implicit-coercion
```
Comment (samturrell) [inline readme.md]: You haven't change the rule name

File: rules/javascript.js [hunk 2]
```diff
@@ -74,5 +77,5 @@ module.exports = {
         'curly': _THROW.WARNING,
         // Encourage using template literals instead of '+' operator on strings
         'prefer-template': _THROW.WARNING,
-    },
-}
+    }
```
Comment (samturrell) [inline rules/javascript.js]: Leave the comma dangle

**Review by samturrell — CHANGES_REQUESTED** [review]

**Review by samturrell — APPROVED** [review]

**Pull request merged**